components/Signuser/Signuser.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState } from "react";
+
+type AuthProps = {
+  type: string;
+};
+
+const AuthForm = ({ type }: AuthProps) => {
+  const [userState, setUserState] = useState({
```

Comment:
Are you sure we need one state to handle both the elements of the form, if we want to handle validations don't you think it would better to have 2 different states for both the fields.
Please provide your inputs

Reply:
Having two states for different fields might be similar to what i did. A single state with fields as it's property also suffice since updation of both doesn't depend on each other.